File: backend/app/schemas/prompt_config.py
"""
提示词配置 Schema

PromptConfig 的 Pydantic 模型定义
"""

from typing import List, Optional

from pydantic import BaseModel, Field

from app.schemas.common import TimestampMixin


class PresetDialogue(BaseModel):
    """预设对话项"""
    role: str = Field(..., description="角色: user 或 assistant")
    content: str = Field(..., description="对话内容")


class PromptConfigBase(BaseModel):
    """提示词配置基础字段"""
    prompt_id: str = Field(..., min_length=1, max_length=64, pattern=r"^[a-z0-9_]+$")
    prompt_name: str = Field(..., min_length=1, max_length=128)
    description: Optional[str] = Field(default=None, description="描述")
    system_prompt: str = Field(..., min_length=1)
    variables: Optional[List[str]] = Field(default_factory=list)
    temperature_override: Optional[float] = Field(default=None, ge=0, le=2)
    is_active: bool = Field(default=True)
    # 工具配置
    selected_tools: Optional[List[str]] = Field(default_factory=list, description="选中的工具ID列表")
    use_all_tools: bool = Field(default=True, description="是否使用全部工具")
    # 预设对话
    preset_dialogues: Optional[List[PresetDialogue]] = Field(default_factory=list, description="预设对话列表")


class PromptConfigCreate(PromptConfigBase):
    """创建提示词配置请求"""
    pass


class PromptConfigUpdate(BaseModel):
    """更新提示词配置请求"""
    prompt_name: Optional[str] = Field(default=None, max_length=128)
    description: Optional[str] = Field(default=None, description="描述")
    system_prompt: Optional[str] = None
    variables: Optional[List[str]] = None
    temperature_override: Optional[float] = Field(default=None, ge=0, le=2)
    is_active: Optional[bool] = None
    selected_tools: Optional[List[str]] = Field(default=None, description="选中的工具ID列表")
    use_all_tools: Optional[bool] = Field(default=None, description="是否使用全部工具")
    preset_dialogues: Optional[List[PresetDialogue]] = Field(default=None, description="预设对话列表")


class PromptConfigInDB(PromptConfigBase, TimestampMixin):
    """数据库中的提示词配置"""
    id: int
    
    class Config:
        from_attributes = True


class PromptConfigList(BaseModel):
    """提示词配置列表项"""
    prompt_id: str
    prompt_name: str
    description: Optional[str] = Field(default=None, description="描述")
    is_active: bool

File: frontend/src/api/promptConfig.ts
import { http } from "./request";

/** 预设对话项 */
export interface PresetDialogue {
  role: "user" | "assistant";
  content: string;
}

/** 提示词配置基础类型 */
export interface PromptConfig {
  id: number;
  prompt_id: string;
  prompt_name: string;
  description?: string;
  system_prompt: string;
  variables?: string[];
  temperature_override?: number;
  is_active: boolean;
  selected_tools?: string[];
  use_all_tools: boolean;
  preset_dialogues?: PresetDialogue[];
  created_at?: string;
  updated_at?: string;
}

/** 提示词配置列表项 */
export interface PromptConfigListItem {
  prompt_id: string;
  prompt_name: string;
  description?: string;
  is_active: boolean;
}

/** 创建提示词配置请求 */
export interface CreatePromptConfigRequest {
  prompt_id: string;
  prompt_name: string;
  description?: string;
  system_prompt: string;
  variables?: string[];
  temperature_override?: number;
  is_active?: boolean;
  selected_tools?: string[];
  use_all_tools?: boolean;
  preset_dialogues?: PresetDialogue[];
}

/** 更新提示词配置请求 */
export interface UpdatePromptConfigRequest {
  prompt_name?: string;
  description?: string;
  system_prompt?: string;
  variables?: string[];
  temperature_override?: number;
  is_active?: boolean;
  selected_tools?: string[];
  use_all_tools?: boolean;
  preset_dialogues?: PresetDialogue[];
}

/** API 响应包装 */
export interface ApiResponse<T> {
  code: number;
  message: string;
  data: T;
}

/**
 * 获取提示词配置列表
 */
export const getPromptConfigList = async (): Promise<
  PromptConfigListItem[]
> => {
  const response = await http.get<ApiResponse<PromptConfigListItem[]>>(
    "/prompt-configs",
  );
  return response.data;
};

/**
 * 获取提示词配置详情
 */
export const getPromptConfigDetail = async (
  promptId: string,
): Promise<PromptConfig> => {
  const response = await http.get<ApiResponse<PromptConfig>>(
    `/prompt-configs/${promptId}`,
  );
  return response.data;
};

/**
 * 创建提示词配置
 */
export const createPromptConfig = async (
  data: CreatePromptConfigRequest,
): Promise<PromptConfig> => {
  const response = await http.post<ApiResponse<PromptConfig>>(
    "/prompt-configs",
    data,
  );
  return response.data;
};

/**
 * 更新提示词配置
 */
export const updatePromptConfig = async (
  promptId: string,
  data: UpdatePromptConfigRequest,
): Promise<PromptConfig> => {
  const response = await http.put<ApiResponse<PromptConfig>>(
    `/prompt-configs/${promptId}`,
    data,
  );
  return response.data;
};

/**
 * 删除提示词配置
 */
export const deletePromptConfig = async (promptId: string): Promise<void> => {
  await http.delete<ApiResponse<void>>(`/prompt-configs/${promptId}`);
};
